dev/ci/check_commands.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+if ! command -v docker &> /dev/null; then
+    echo "docker could not be found, exiting."
+    exit 1
+fi
+
+if ! command -v docker-proxy &> /dev/null; then
```

Comment:
Should you also check if it is executable to handle the issue you met?

Reply:
I'm not very confident that this is a common occurrence. Since it happened so recently, I will add checks for it.